padle/padle.py

```diff
@@ -0,0 +1,489 @@
+            return await send_cancellation_message(ctx, "You have already finished today's PADle!")
+        async with self.config.user(ctx.author).todays_guesses() as todays_guesses:
+            todays_guesses.append(guess_monster.monster_id)
+        monster = dbcog.get_monster(int(await self.config.padle_today()))
+        cur_day = await self.config.num_days()
+        todays_guesses = await self.config.user(ctx.author).todays_guesses()
+        async with self.config.user(ctx.author).all_guesses() as all_guesses:
+            all_guesses[str(cur_day)] = todays_guesses
+
+        try:
+            channel = self.bot.get_channel(await self.config.user(ctx.author).channel_id())
+            del_msg = await channel.fetch_message(await self.config.user(ctx.author).edit_id())
+            await del_msg.delete()
+        except discord.NotFound:  # user already deleted message
+            pass
+
+        guess_monster_diff = MonsterDiff(monster, guess_monster)
+        points = guess_monster_diff.get_diff_score()
+
+        padle_menu = PADleScrollMenu.menu()
+        # query_settings = await QuerySettings.extract_raw(ctx.author, self.bot, guess)
+        cur_page = ceil(len(todays_guesses) / 5) - 1
+        page_guesses = await PADleScrollViewState.do_queries(dbcog, todays_guesses)
+        state = PADleScrollViewState(ctx.author.id, PADleScrollMenu.MENU_TYPE, guess, monster=monster,
+                                     cur_day_guesses=page_guesses, current_day=cur_day,
+                                     current_page=cur_page)
+        message = await padle_menu.create(ctx, state)
+        await self.config.user(ctx.author).edit_id.set(message.id)
+        await self.config.user(ctx.author).channel_id.set(ctx.channel.id)
+
+        score = "\N{LARGE YELLOW SQUARE}"
+        if guess_monster.monster_id == monster.monster_id:
+            await ctx.send("You got the PADle in {} guesses! Use `{}padle score` to share your score.".format(
+                len(await self.config.user(ctx.author).todays_guesses()), prefix))
+            await self.config.user(ctx.author).done.set(True)
+            async with self.config.all_scores() as all_scores:
+                all_scores.append(str(len(await self.config.user(ctx.author).todays_guesses())))
+            score = "\N{LARGE GREEN SQUARE}"
+        else:
+            if points < 9:
+                score = "\N{LARGE ORANGE SQUARE}"
+            if points < 5:
+                score = "\N{LARGE RED SQUARE}"
+        async with self.config.user(ctx.author).score() as scores:
+            scores.append(score)
+        # start_adding_reactions(message, emojis)
+        # await menus.menu(ctx, embed_pages, embed_controls, message=message, page=len(embed_pages) - 1, timeout=60 * 60)
+
+    @padle.command()
+    async def giveup(self, ctx):
+        """Give up on today's PADle"""
+        prefix = ctx.prefix
+        if ctx.guild is not None and not await self.can_play_in_guild(ctx):
```

Comment:
You could pull out all of these "check there's actually a game & they're playing & etc into a single function called like `def ensure_active_game` or whatever and then run that at the start of each function that requires an active game